docs/tsp/typeServerProtocol.ts

```diff
@@ -0,0 +1,383 @@
+        GetUnionTypes = 'typeServer/getUnionTypes',
+        // Request to find a member of a type.
+        GetTypeMember = 'typeServer/getTypeMember',
+        // Request to get all overloads of a function or method.
+        GetOverloads = 'typeServer/getOverloads',
+        // Request to get the effective meta class of a type.
+        GetEffectiveMetaClass = 'typeServer/getEffectiveMetaClass',
+        // Request to get the type of a declaration.
+        GetTypeOfDeclaration = 'typeServer/getTypeOfDeclaration',
+        // Request to get symbol declaration information for a node.
+        GetSymbolDeclarationInfo = 'typeServer/getSymbolDeclarationInfo',
+        // Request to get the bound magic method for a type.
+        GetBoundMagicMethod = 'typeServer/getBoundMagicMethod',
+        // Request to resolve an alias declaration.
+        // This request is sent by the client to resolve an alias declaration, which may involve looking up the
+        // declaration in the type server's cache or performing additional resolution logic.
+        ResolveAliasDeclaration = 'typeServer/resolveAliasDeclaration',
+        // Request to resolve an import.
+        ResolveImport = 'typeServer/resolveImport',
+    }
+
+    export enum Notifications {
+        // Notification sent by the server to indicate that the type server has been initialized.
+        Initialized = 'typeServer/initialized',
+        // Notification sent by the server to indicate that the type server has shut down.
+        Shutdown = 'typeServer/shutdown',
+        // Notification sent by the client to indicate that a text document has been opened.
+        TextDocumentOpen = 'textDocument/open',
+        // Notification sent by the client to indicate that a text document has been closed.
+        TextDocumentClose = 'textDocument/close',
+        // Notification sent by the client to indicate that settings have changed.
+        SettingsChange = 'typeServer/settingsChange',
+        // Notification sent by the client to indicate that files have changed.
+        FilesChanged = 'typeServer/filesChanged',
+        // Notification sent by the client to indicate that the type server's cache should be flushed (and snapshots reset)
+        FlushCache = 'typeServer/flushCache',
+    }
+
+    export interface Params {
+        [Requests.Initialize]: { workspace: WorkspaceFolder; initialSettings: string };
+        [Requests.Shutdown]: void;
+        [Requests.GetSnapshot]: void;
+        [Requests.GetDiagnostics]: { uri: string; snapshot: number };
+        [Requests.GetDiagnosticsVersion]: { uri: string };
+        [Requests.GetType]: { node: Node; snapshot: number };
+        [Requests.GetUnionTypes]: { type: Type; snapshot: number };
+        [Requests.GetTypeMember]: {
+            type: Type;
+            memberName: string;
+            accessFlags: MemberAccessFlags;
+            snapshot: number;
+        };
+        [Requests.GetOverloads]: { type: Type; snapshot: number };
```

Comment:
Should `GetOverloads` return the overload implementation signature as well as the overloads? Or should it filter out the overload implementation signature?

Reply:
It's probably better if it is documented to return only the overloaded signatures (i.e. filter them out). This could help facilitate collapsing the `Function` and `Overloaded` categories into a single category.